src/precompiles/modexp.rs

```diff
@@ -132,24 +66,96 @@ impl Precompile for ModExp<Byzantium> {
             }
         };
 
-        Ok(PrecompileOutput::without_logs(cost, output))
+        Ok(output)
+    }
+}
+
+impl ModExp<Byzantium> {
+    fn mul_complexity(x: u64) -> Result<u64, ExitError> {
+        if x <= 64 {
+            Ok(x * x)
+        } else if x <= 1_024 {
+            Ok(x * x / 4 + 96 * x - 3_072)
+        } else {
+            let (x_sq, overflow) = x.overflowing_mul(x);
+            if overflow {
+                Err(ExitError::OutOfGas)
```

Comment:
In [the Go implementation](https://github.com/ethereum/go-ethereum/blob/master/core/vm/contracts.go#L295) they use big ints all the way through and then return a `u64` at the end (returning `u64::MAX` if the result would be larger than a `u64` can represent). Perhaps we should do the same.

Reply:
That is what I had originally done, not sure why I opted to remove this in the end.

Reply:
Done here https://github.com/aurora-is-near/aurora-engine/pull/138/commits/29e4336bacd886b3818c6ba6f0be730e16e242f6